lib/classes/Variables.js

```diff
@@ -311,7 +311,14 @@ class Variables {
     const variableStringWithoutSource = variableString.split(':')[1].split('.');
     const stackName = variableStringWithoutSource[0];
     const outputLogicalId = variableStringWithoutSource[1];
-    return this.serverless.getProvider('aws')
+    if (
+      this.cfVars &&
+      this.cfVars[stackName] &&
+      this.cfVars[stackName][outputLogicalId]
+    ) {
+      return this.cfVars[stackName][outputLogicalId];
```

Comment:
You should return a promised value here as the function's current signature also returns promises.
`return BbPromise.resolve(this.cfVars[stackName][outputLogicalId]);`

Reply:
The cached values are promises so I don't think this is needed.